### ISSUE TITLE
Share password(s) silently removed when password policy is changed

### ISSUE THREAD

**Issue by MaxMaxtor** [nextcloud/server#15546]
Only my second bug report. So hopefully I have reported correctly. I did not find the bug in a search. Any questions, please ask :)

### Steps to reproduce
As Admin go to: Setting ---> Security then the "Password policy" section

If no boxes are checked in this section and you then check an option like "Forbid common passwords" and you have existing passwords protecting shares that match this criteria then those existing passwords are silently removed from protecting the shares. The previously password protected shares are now accessible without a password.

### Expected behaviour
Some warning of shares that have these insecure passwords so they can be changed or similar options.

### Actual behaviour
If no boxes are checked in this section and you then check an option like "Forbid common passwords" and you have existing passwords protecting shares that match this criteria then those existing passwords are silently removed from protecting the shares. The previously password protected shares are now accessible without a password.

## Server configuration detail

**Operating system:** Linux 4.14.98-v7+ #1200 SMP Tue Feb 12 20:27:48 GMT 2019 armv7l

**Webserver:** Apache/2.4.25 (Raspbian) (apache2handler)

**Database:** mysql 10.1.38

**PHP version:** 

7.0.33-0+deb9u3
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, apc, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, igbinary, imagick, intl, json, exif, mcrypt, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 15.0.7 - 15.0.7.0

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.1.0
 - activity: 2.8.2
 - bruteforcesettings: 1.3.0
 - calendar: 1.6.5
 - cloud_federation_api: 0.1.0
 - comments: 1.5.0
 - contacts: 3.1.1
 - dav: 1.8.1
 - federatedfilesharing: 1.5.0
 - federation: 1.5.0
 - files: 1.10.0
 - files_pdfviewer: 1.4.0
 - files_sharing: 1.7.0
 - files_texteditor: 2.7.0
 - files_trashbin: 1.5.0
 - files_versions: 1.8.0
 - files_videoplayer: 1.4.0
 - firstrunwizard: 2.4.0
 - gallery: 18.2.0
 - issuetemplate: 0.5.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.3.0
 - nextcloud_announcements: 1.4.0
 - notifications: 2.3.0
 - oauth2: 1.3.0
 - password_policy: 1.5.0
 - provisioning_api: 1.5.0
 - serverinfo: 1.5.0
 - sharebymail: 1.5.0
 - spreed: 5.0.3
 - support: 1.0.0
 - survey_client: 1.3.0
 - systemtags: 1.5.0
 - theming: 1.6.0
 - twofactor_backupcodes: 1.4.1
 - updatenotification: 1.5.0
 - workflowengine: 1.5.0
Disabled:
 - admin_audit
 - encryption
 - files_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "192.168.6.200",
        "localhost",
        "rahtech.co.uk"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "15.0.7.0",
    "overwrite.cli.url": "http:\/\/192.168.1.252\/nextcloud",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "mail_smtpmode": "smtp",
    "mail_smtpsecure": "ssl",
    "mail_smtpauthtype": "LOGIN",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "25",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "filelocking.enabled": "true",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "timeout": 0
    },
    "maintenance": false,
    "theme": "",
    "loglevel": 2,
    "mail_sendmailmode": "smtp",
    "mysql.utf8mb4": true
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) Gecko/20100101 Firefox/66.0

**Operating system:** Ubuntu 18.04.2 LTS

## Logs

<details><summary>Web server error log</summary>

```
Insert your web server log here 
```
</details>

<details><summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

<details><summary>Browser log</summary>

Insert your browser log here, this could for example include:

	a) The javascript console log
	b) The network log
	c) ...
</details>

**Comment by skjnldsv** [comment]
> silently removed from protecting the shares

Hey! When you say silently removed, you mean in the database?

**Comment by MaxMaxtor** [comment]
Hi skjnldsv :)

I just check the issue this morning after your message and I can not replicate the issue now. All seems to be working as expected. 

What happened to me originally was that I shared a document with a simple password via  a link created by Nextcloud. Later I changed the password security settings for strength in the Nextcloud admin area. When I then visited the link of the previously passworded document that did not reach the new minimum password requirements I was able to access the share with out being prompted for the password. I originally tested in container tabs and different browsers to make sure that my browser was not storing anything that was letting me back in to the share without the password prompt. May be I was mistaken or something has changed since Nextcloud version: 15.0.7 - 15.0.7.0 either way it all seems fine now.
